core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -55,72 +56,80 @@ func getCommitMethodConfig(fromAddress string, routerProgramAddress string, comm
 				IsWritable: true,
 			},
 			chainwriter.PDALookups{
-				Name: "RouterReportAccount",
-				PublicKey: chainwriter.AccountConstant{
-					Address:    routerProgramAddress,
-					IsSigner:   false,
-					IsWritable: false,
-				},
+				Name:      "CommitReport",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
 				Seeds: []chainwriter.Seed{
 					{Static: []byte("commit_report")},
 					{Dynamic: chainwriter.AccountLookup{Location: merkleRootChainSelector}},
-					{Dynamic: chainwriter.AccountLookup{
-						Location: "Info.MerkleRoots.MerkleRoot",
-					}},
+					{Dynamic: chainwriter.AccountLookup{Location: merkleRoot}},
 				},
 				IsSigner:   false,
-				IsWritable: false,
+				IsWritable: true,
 			},
 			getAuthorityAccountConstant(fromAddress),
 			getSystemProgramConstant(),
-			chainwriter.AccountConstant{
-				Name:       "SysvarInstructions",
-				Address:    sysvarInstructionsAddress,
-				IsSigner:   true,
+			getSysVarInstructionConstant(),
+			getFeeBillingSignerConfig(offrampProgramAddress),
+			getFeeQuoterConfig(offrampProgramAddress),
+			chainwriter.PDALookups{
+				Name: "FeeQuoterAllowedPriceUpdater",
+				// Fetch fee quoter public key to use as program ID for PDA
+				PublicKey: getFeeQuoterConfig(offrampProgramAddress),
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("allowed_price_updater")},
+					{Dynamic: getFeeBillingSignerConfig(offrampProgramAddress)},
+				},
+				IsSigner:   false,
 				IsWritable: false,
 			},
 			chainwriter.PDALookups{
-				Name: "GlobalState",
-				PublicKey: chainwriter.AccountConstant{
-					Address: routerProgramAddress,
+				Name: "FeeQuoterConfig",
+				// Fetch fee quoter public key to use as program ID for PDA
+				PublicKey: getFeeQuoterConfig(offrampProgramAddress),
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("config")},
 				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
```

Comment:
Can we confirm whether these last 3 accounts are indeed optional? Cc: @toblich

Reply:
Seems like they are since they aren't needed if there's no token price or gas price update [here](https://github.com/smartcontractkit/chainlink-ccip/blob/d9b9b053905ed73dd128cd51896dbe85b344dbde/chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/commit.rs#L43-L51)